tests/Client/Client12RequestResponsesTest.php

```diff
@@ -398,4 +402,36 @@ public function testApplicationError(int $code): void
             '), $detail->asXML());
         }
     }
+
+    public function testSerializerContextParametersAreAdded()
+    {
+        // Override serializer with a custom handler for asserting the context parameters
+        $handlers = function (HandlerRegistryInterface $h) {
```

Comment:
instead of rebuilding from scratch the serializer instance for this single test, why not expose the handler registry to the test case class?


In https://github.com/goetas-webservices/soap-client/blob/9dfeb8198f986c201c21a5fe7b4049a1baa93187/tests/Client/RequestResponsesTest.php#L84

```php

     protected $handlerRegistry;
    //.....
   function setup(){
   //...
        $handlers = static function (HandlerRegistryInterface $h) use ($headerHandler): void {
            $this->handlerRegistry = $h;
            $h->registerSubscribingHandler($headerHandler);
        };
  }
```


so later you can register your custom handler using `$this->handlerRegistry->registerHandler(...)`

Reply:
Thanks, good tip.